v-next/hardhat/test/internal/builtin-plugins/solidity/build-system/build-system.ts

```diff
@@ -55,43 +54,76 @@ describe(
     skip: process.env.HARDHAT_DISABLE_SLOW_TESTS === "true",
   },
   () => {
-    let artifactsPath: string;
-    let hre: HardhatRuntimeEnvironment;
+    let actualArtifactsPath: string;
+    let actualCachePath: string;
+    let expectedArtifactsPath: string;
+    let expectedCachePath: string;
+    let solidity: SolidityBuildSystemImplementation;
 
     useFixtureProject("solidity/example-project");
 
-    beforeEach(async () => {
-      const tmpDir = await getTmpDir("solidity-build-system-implementation");
-      artifactsPath = path.join(tmpDir, "artifacts");
-      const cachePath = path.join(tmpDir, "cache");
-      hre = await createHardhatRuntimeEnvironment({
-        paths: {
-          artifacts: artifactsPath,
-          cache: cachePath,
-        },
-        solidity: {
-          profiles: {
-            default: {
-              compilers: [
-                {
-                  version: "0.8.22",
-                },
-                {
-                  version: "0.7.1",
-                },
-              ],
+    const solidityConfig: SolidityConfig = {
+      profiles: {
+        default: {
+          compilers: [
+            {
+              version: "0.8.22",
+              settings: {},
             },
-          },
-          remappings: ["remapped/=npm/@openzeppelin/contracts@5.1.0/access/"],
+            {
+              version: "0.7.1",
+              settings: {},
+            },
+          ],
+          overrides: {},
         },
+      },
+      dependenciesToCompile: [],
+      remappings: ["remapped/=npm/@openzeppelin/contracts@5.1.0/access/"],
+    };
+
+    before(async () => {
+      expectedArtifactsPath = path.join(process.cwd(), "artifacts");
+      expectedCachePath = path.join(process.cwd(), "cache");
+      await remove(expectedArtifactsPath);
+      await remove(expectedCachePath);
+      const hooks = new HookManagerImplementation(process.cwd(), []);
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- We don't care about hooks in this context
+      hooks.setContext({} as HookContext);
```

Comment:
When we call `build()` it is asserted that the hook context is not `undefined`. We don't actually use hooks in this tests.

A proper `HookContext` contains `hre`. I didn't want to create a full `hre` for this test. Hence, the type cast.

Reply:
ok for now, but this will break soon, when we add hooks to the build system.